crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,561 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+/// Grouping of core control parameter of the layout for this node.
+#[derive(Component, Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect(PartialEq, Serialize, Deserialize)]
+pub struct LayoutControl {
```

Comment:
If this is the "universal layout configuration component", maybe we can shorten the name to `Layout`? Like `Display` is really the "layout method selector", but we're keeping it aligned with CSS terminology, so that frees up the `Layout` term. This component encapsulates the "canonical layout method selector", so `Layout` feels like fair game.